plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java

```diff
@@ -30,31 +39,80 @@ public class TransportPredictionTaskAction extends HandledTransportAction<Action
     TransportService transportService;
     MLModelCacheHelper modelCacheHelper;
 
+    Client client;
+
+    ClusterService clusterService;
+
+    NamedXContentRegistry xContentRegistry;
+
+    MLModelManager mlModelManager;
+
+    ModelAccessControlHelper modelAccessControlHelper;
+
     @Inject
     public TransportPredictionTaskAction(
         TransportService transportService,
         ActionFilters actionFilters,
         MLPredictTaskRunner mlPredictTaskRunner,
-        MLModelCacheHelper modelCacheHelper
+        MLModelCacheHelper modelCacheHelper,
+        ClusterService clusterService,
+        Client client,
+        NamedXContentRegistry xContentRegistry,
+        MLModelManager mlModelManager,
+        ModelAccessControlHelper modelAccessControlHelper
     ) {
         super(MLPredictionTaskAction.NAME, transportService, actionFilters, MLPredictionTaskRequest::new);
         this.mlPredictTaskRunner = mlPredictTaskRunner;
         this.transportService = transportService;
         this.modelCacheHelper = modelCacheHelper;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+        this.mlModelManager = mlModelManager;
+        this.modelAccessControlHelper = modelAccessControlHelper;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLTaskResponse> listener) {
         MLPredictionTaskRequest mlPredictionTaskRequest = MLPredictionTaskRequest.fromActionRequest(request);
         String modelId = mlPredictionTaskRequest.getModelId();
-        String requestId = mlPredictionTaskRequest.getRequestID();
-        log.debug("receive predict request " + requestId + " for model " + mlPredictionTaskRequest.getModelId());
-        long startTime = System.nanoTime();
-        mlPredictTaskRunner.run(mlPredictionTaskRequest, transportService, ActionListener.runAfter(listener, () -> {
-            long endTime = System.nanoTime();
-            double durationInMs = (endTime - startTime) / 1e6;
-            modelCacheHelper.addPredictRequestDuration(modelId, durationInMs);
-            log.debug("completed predict request " + requestId + " for model " + modelId);
-        }));
+
+        User user = mlPredictionTaskRequest.getUser();
+        if (user == null) {
+            user = RestActionUtils.getUserContext(client);
+            mlPredictionTaskRequest.setUser(user);
+        }
+        final User userInfo = user;
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            mlModelManager.getModel(modelId, ActionListener.wrap(mlModel -> {
```

Comment:
User may concern about the overhead of querying model and model group index. How about we skip access checking if user don't enable backend role filtering?

Reply:
This is some performance enhancement. We can do this later.

Reply:
Yeah agreed, this can improve performance